etl/steps/data/garden/countries/2023-09-25/shared.py

```diff
@@ -91,3 +93,44 @@ def _get_end_year(date_str: str, date_format: str) -> int:
     if (date.month == 12) & (date.day == 31):
         return date.year + 1
     return date.year
+
+
+def add_population_to_table(
+    tb: Table, ds_pop: Dataset, country_col: str = "country", region_alt: bool = False
+) -> Table:
+    """Bla.
+
+    1. Get list of countries from latest available year. That is, we only have one row per country.
+    2. Duplicate these entries for each year from first available to latest available year. As if they existed.
+        This is because the population dataset tracks population back in time with current countries' borders.
+    3. Merge with population dataset
+    """
+    YEAR_MAX = tb["year"].max()
+    YEAR_MIN = tb["year"].min()
+    # Get last year data
+    tb_last = tb[tb["year"] == YEAR_MAX].drop(columns=["year"])
+
+    # Extend to all years
+    tb_all_years = Table(pd.RangeIndex(YEAR_MIN, LAST_YEAR + 1), columns=["year"])
+    tb_pop = tb_last.merge(tb_all_years, how="cross")
+
+    # Add population
+    tb_pop = geo.add_population_to_table(tb_pop, ds_pop, country_col=country_col)
+
+    # Estimate population by region
+    tb_pop_regions = tb_pop.groupby(["year", "region"], as_index=False)[["population"]].sum()
```

Comment:
How about using the [add_region_aggregates function](https://github.com/owid/etl/blob/f320159a10202a2064c3149a6e09e2ae00fc2329/etl/data_helpers/geo.py#L231)? It can be useful to not generate aggregations in case of low coverage

Reply:
Thanks for proposing, Pablo! The thing with this dataset is a bit particular about their regions and their member states. I use a `if-else` like clause to assign countries to regions based on their country codes. Doing it like this is slightly easier in this case.